src/components/carousel/carousel.hook.ts

```diff
@@ -0,0 +1,36 @@
+import { useState } from 'react'
+import { useSwipeable } from 'react-swipeable'
+import { SlideData } from './carousel.types'
+
+enum UPDATE_COUNT {
+  INCREMENT,
+  DECREMENT,
+}
+
+export function useCarousel(slidesData: SlideData[]) {
+  const [activeIndex, setActiveIndex] = useState<number>(0)
+
+  const handleSwipe = (updateCount: UPDATE_COUNT) => {
+    if (
+      (updateCount === UPDATE_COUNT.DECREMENT && activeIndex > 0) ||
+      (updateCount === UPDATE_COUNT.INCREMENT &&
+        activeIndex < slidesData.length - 1)
+    ) {
+      const newActiveIndex = activeIndex + updateCount
+      setActiveIndex(newActiveIndex)
+    }
+  }
```

Comment:
acho que essa lógica está um tanto confusa, não seria algo do tipo:

```suggestion
  const handleSwipe = (updateCount: UPDATE_COUNT) => {
    if (updateCount === UPDATE_COUNT.DECREMENT ) {
      activeIndex === 0 ? 
        setActiveIndex(() => slidesData.length - 1) :
        setActiveIndex(oldIndexState => oldIndexState - 1)
    } else {
      activeIndex === slidesData.length - 1 ? 
        setActiveIndex(0) :
        setActiveIndex(oldIndexState => oldIndexState + 1)
    }
  }
```

Reply:
Arrumei